Fix most of the console errors

Arthurk12 opened the pull request:
### What does this PR do?
This PR addresses multiple console errors that appeared in the development environment. For example, simply loading the client previously triggered 26+ warnings and errors in the browser console.
While this PR doesn't eliminate all console errors, it significantly reduces the noise and addresses the majority of them.
Remaining issues stem from third-party libraries:
- TlDraw
  > react-dom.development.js:86 Warning: Cannot update a component (`ShapesToDisplay`) while rendering a different component (`Layout`). To locate the bad setState() call inside `Layout`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
- Draggable
  > Warning: findDOMNode is deprecated and will be removed in the next major release. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-nodeStack
- Etherpad
  > Uncaught ReferenceError: require is not defined
    at ace2_common.js?callb…fine&v=e74a56ce:1:1
- Tabs
  > Warning: Tab: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.

Arthurk12 committed:
fix: app component bad set state

Changes logOnJoin into a reference, since it was being called inside
componentDidUpdate, which is not allowed - called during render.

Arthurk12 committed:
fix(hooks): fix bad state caused by deduplicated subscription

Rewrite the deduplicated subscription hook to encapsulate side effects
within `useEffect` blocks, avoiding invalid state transitions.

Arthurk12 committed:
fix(presentation): replace deprecated defaultProps usage

Replaces `defaultProps` with default parameter values in function argument
destructuring, aligning with modern React and functional component patterns.

Arthurk12 committed:
fix(base-button): 'loading' and 'hoverColor' props

Arthurk12 committed:
fix(media-area): replace deprecated defaultProps usage

Replaces `defaultProps` with default parameter values in function argument
destructuring, aligning with modern React and functional component patterns.

Arthurk12 committed:
fix(reactions): missing 'onClick' event handler

Arthurk12 committed:
fix(tooltip): add 'right' and 'left' as possible positions

Adds 'right' and 'left' positions to the tooltip component since the
underlying lib - Tippy - accepts that placement for the tooltip.

Arthurk12 committed:
fix(presentation): correctly define 'currentSlide' and
'currentPresentationPage'

Correctly define the datatype of 'currentSlide' and
'currentPresentationPage'.

Arthurk12 committed:
fix(reactions): replace findDOMNode by reference

Arthurk12 committed:
fix(user-list): missing labels in crowd actions button

Arthurk12 committed:
fix(user-list): correct type for children

Changes the type for 'children' prop in User Avatar component.

Arthurk12 committed:
fix(presentation): correct prop 'isDarkThemeEnabled'

Arthurk12 committed:
fix(nav-bar): removes unused props

Arthurk12 committed:
refactor(app): convert files to typescript

Convert app container and component to typescript. Also took the
oportunity to refactor some variables and props while defining their
types.

Arthurk12 committed:
fix(apps-gallery): missing label and onClick properties

Arthurk12 committed:
fix(breakouts): missing label property on buttons

Arthurk12 committed:
fix(video-list): fix 'cameraId' prop type

Arthurk12 committed:
fix(presentation): bad state in uploader toast

Wraps side-effects in useEffect hook to prevent bad state error in react
during render of presentation uploader toast.

Arthurk12 committed:
fix(tests): correct selector for 'managePresentations'

Arthurk12 committed:
fix(presentation): restore presentation on update

Get the restore presentation on update from the correct place.

Arthurk12 committed:
fix(presentation): update presentation size regardless of slide change

The function to update the presentation size was only executed when
there was a slide change. This prevented the presentation to occupy the
maximum available space in some scenarios where the presentation sizes
update came in different renders from slide change.

Arthurk12 committed:
fix(tests): update snapshots for PRESENTATION ONLY

germanocaumo commented:
Just reporting (not sure it's the same the tabs one): still seeing some warnings when opening the share camera preview modal:

```
hook.js:608 Warning: Failed prop type: There should be an equal number of 'Tab' and 'TabPanel' in `Tabs`. Received 2 'Tab' and 0 'TabPanel'.
hook.js:608 Warning: Failed prop type: There should be an equal number of 'Tab' and 'TabPanel' in `UncontrolledTabs`. Received 2 'Tab' and 0 'TabPanel'.
```

Arthurk12 committed:
chore(ts): add types to queries and ignore untyped JS components

Define return types for subscriptions using existing or new interfaces.
Suppress type errors caused by the UserAvatar component, which is still in JSX,
by adding @ts-ignore where necessary.

Arthurk12 committed:
chore: add types package for 'react-modal'

Arthurk12 committed:
fix(captions): enhance accessibility across devices for live
transcription

Updates the captions wrapper to use section role and aria-label instead
of role region as suggested by SonarCloud.

Arthurk12 committed:
fix: replace deprecated `KeyboardEvent.keyCode` property

The `keyCode` property of keyboard events has been deprecated (see link
1). This commit replaces all `event.keyCode` occurrences with `event.key`.
It also standardizes the removal of other deprecated properties such as
`event.which`(see link 2). The `keycodes` file was removed and replaced by
an updated list of key constants that match the values returned by
`event.key`(see link 3 and 4).

Links:
1: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
2: https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/which
3: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key
4: https://caniuse.com/keyboardevent-key

Arthurk12 committed:
fix: whiteboard shortcuts and unused prop removal

Fix an issue where the undo/redo shortcuts on the whiteboard were not
working due to how the `event.key` attribute returns pressed keys —
it returns uppercase characters when the Shift key is pressed, and
lowercase otherwise.
Additionally, remove an unused prop from object destructuring.

Arthurk12 committed:
fix(tests): whiteboard shape duplication shortcut

Adjust the pressed keys on the test to lowercase 'd', since the `event.key`
attribute differentiates upper and lowercase letters.
